### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by muskaanshah**
Fetaures added:
Theme change option
Basic authentication
Redirecting user when clicked on private routes

### CODE REVIEW

**Review by manismk — APPROVED** [review]
Functionality works properly.
Good naming convention and code is easy to understand.

File: src/backend/db/users.js [hunk 1]
```diff
@@ -7,13 +7,13 @@ import { formatDate } from "../utils/authUtils";
...
-    _id: uuid(),
-    firstName: "Adarsh",
-    lastName: "Balika",
-    email: "adarshbalika@gmail.com",
-    password: "adarshBalika123",
-    createdAt: formatDate(),
-    updatedAt: formatDate(),
-  },
+    {
+        _id: uuid(),
```
Comment (shobit1337) [inline src/backend/db/users.js:11]: Here, try to use a static id in mockbee,

File: src/components/Drawer/Drawer.jsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { NavLink } from "react-router-dom";
 import "./drawer.css";
 
 function Drawer({drawer, setDrawer}) {
+	// const {theme} = useTheme();
```
Comment (shobit1337) [inline src/components/Drawer/Drawer.jsx]: remove unwanted commented code

File: src/context/auth-context.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext, useState } from "react";
+
+const authContext = createContext();
```
Comment (shobit1337) [inline src/context/auth-context.js]: Start the name of context with capital letter. its a good practice

File: src/pages/authentication/Login.jsx [hunk 4]
```diff
@@ -1,45 +1,92 @@
...
+        <div className="container-body color-white centered">
+            <div className="card-authentication bg-grey-dark borderradius-0-5 mt-2">
+                <p className="text-center fw-500 fs-1-25">Login</p>
+                <form onSubmit={submitFormHandler}>
+                    <span className="input-box">
+                        <input
+                            type="email"
+                            id="email-address"
+                            className="input-text input-authentication p-0-5 fs-0-9 mb-1"
+                            placeholder="Email Address *"
```
Comment (shobit1337) [inline src/pages/authentication/Login.jsx:37]: You can add checks here for valid email

File: src/services/authServices.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+const loginUser = async (
+    email,
+    password,
+    setToken,
+    setUser,
+    setError,
+    navigate,
+    location
+) => {
+    let from = location.state?.from?.pathname || "/";
```
Comment (shobit1337) [inline src/services/authServices.js]: try using const as much as possible

**Review by shobit1337 — APPROVED** [review]

**Comment by shobit1337** [comment]
I've suggested a few suggestions, Rest everything looks good!

**Pull request merged**